src/main/java/com/conveyal/gtfs/model/Transfer.java

```diff
@@ -99,7 +118,20 @@ protected void writeOneRow(Transfer t) throws IOException {
         protected Iterator<Transfer> iterator() {
             return feed.transfers.values().iterator();
         }
+    }
 
-
+    /**
+     * Transfer entries have no ID in GTFS so we define one based on the fields in the transfer entry.
+     */
+    private static String createId(
+        String fromStopId,
+        String toStopId,
+        String fromTripId,
+        String toTripId,
+        String fromRouteId,
+        String toRouteId
+    ) {
+        return String.format("%s_%s_%s_%s_%s_%s", fromStopId, toStopId, fromTripId, toTripId, fromRouteId, toRouteId);
```

Comment:
I suppose this supports null values on some of the optional fields.

Reply:
It produces for example: `4u6g_123_null_null_null_null`, which will still be unique with other rows that have not defined optional fields providing the from and to stop ids are unique (which would flag a duplicate error if not).